### ISSUE TITLE
Chatbot breaks if update its content too quickly

### ISSUE THREAD

**Issue by GaiZhenbiao** [gradio-app/gradio#7970]
### Describe the bug

If you update the Chatbot component too quickly, it errors out, producing strange outputs. I suspect the issue isn't limited to the Chatbot component alone; it might be that Gradio's underlying data flow isn't thread-safe or something.

No console error message is printed when this error is triggered with the reproduction code. However, in my real application, I did see an error message that directly corresponds to this issue, see below. If you're wondering, no, the reason the reproduction code didn't print an error message like this isn't due to logging level settings. I haven't figured out why.

I'm using the latest version of Gradio. I can reproduce this issue 100% of the time on Chromium and Safari, Firefox seems fine for some reason.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import json
import time

import gradio as gr

with open("chatbot.json", "r") as f:
    data = json.load(f)

def update_chatbot():
    for d in data:
        time.sleep(0.005)
        yield d


with gr.Blocks() as demo:
    chatbot = gr.Chatbot()

    trigger_btn = gr.Button("Trigger Chatbot Bug")
    trigger_btn.click(update_chatbot, [], chatbot)

demo.queue().launch()
```
[chatbot.json](https://github.com/gradio-app/gradio/files/14908910/chatbot.json)

### Screenshot

Arc (Chromium):
![a55d02208244befd2e92d3b49e72cb0e](https://github.com/gradio-app/gradio/assets/51039745/b4dc9266-e1c0-4487-9a53-fe299246ccf6)
Safari:
<img width="1436" alt="image" src="https://github.com/gradio-app/gradio/assets/51039745/72c2eff8-ec60-4b9c-8ab1-f352e2584018">
Firefox seems fine though.
<img width="1263" alt="image" src="https://github.com/gradio-app/gradio/assets/51039745/f1e8c4f5-d4dc-4e95-8691-e86ead94c214">
Browser console:
<img width="419" alt="image" src="https://github.com/gradio-app/gradio/assets/51039745/9de30ead-d9b8-40f9-82bd-3e00170dc198">

### Logs

```shell
Traceback (most recent call last):
  File "/opt/homebrew/anaconda3/envs/latest_chuanhu/lib/python3.11/site-packages/gradio/queueing.py", line 522, in process_events
    response = await route_utils.call_process_api(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/latest_chuanhu/lib/python3.11/site-packages/gradio/route_utils.py", line 260, in call_process_api
    output = await app.get_blocks().process_api(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/latest_chuanhu/lib/python3.11/site-packages/gradio/blocks.py", line 1739, in process_api
    inputs = self.preprocess_data(fn_index, inputs, state, explicit_call)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/latest_chuanhu/lib/python3.11/site-packages/gradio/blocks.py", line 1461, in preprocess_data
    inputs_cached = block.data_model(root=inputs_cached)  # type: ignore
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/latest_chuanhu/lib/python3.11/site-packages/pydantic/root_model.py", line 63, in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(root, self_instance=__pydantic_self__)
pydantic_core._pydantic_core.ValidationError: 3 validation errors for ChatbotData
0.1.str
  Input should be a valid string [type=string_type, input_value=[2, 1], input_type=list]
    For further information visit https://errors.pydantic.dev/2.5/v/string_type
0.1.FileMessage
  Input should be a valid dictionary or instance of FileMessage [type=model_type, input_value=[2, 1], input_type=list]
    For further information visit https://errors.pydantic.dev/2.5/v/model_type
0
  Tuple should have at most 2 items after validation, not 3 [type=too_long, input_value=['replace', [2, 1], '<div...ce, which is a\n</div>'], input_type=list]
    For further information visit https://errors.pydantic.dev/2.5/v/too_long
```

### System Info

```shell
gradio version: 4.25.0
gradio_client version: 0.15.0
Python version: 3.11.3

OS: macOS Sonoma 14.4.1 ARM64
Browser: Arc 1.37.0 (Chromium), Safari version 17.5 (19618.2.4.11.2)
```

### Severity

Blocking usage of gradio

**Comment by freddyaboulton** [comment]
I believe this has been fixed by #7926 and will be shipped in the next release 4.26.0 shortly. Can re-open if this issue is not fixed but I just tested `main` against your repro and the bug does not happen.

**Comment by keenranger** [comment]
@freddyaboulton This bug does not happend to me after `4.26.0` release. Thankyou!
